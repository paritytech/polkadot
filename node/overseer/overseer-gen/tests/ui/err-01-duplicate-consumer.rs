#![allow(dead_code)]

use polkadot_overseer_gen::*;

#[derive(Default)]
struct AwesomeSubSys;

#[derive(Default)]
struct AwesomeSubSys2;

#[derive(Clone, Debug)]
struct SigSigSig;

struct Event;

#[derive(Clone)]
struct MsgStrukt(u8);

#[overlord(signal=SigSigSig, event=Event, gen=AllMessages, error=OverseerError)]
struct Overseer {
	#[subsystem(MsgStrukt)]
	sub0: AwesomeSubSys,

	#[subsystem(MsgStrukt)]
	sub1: AwesomeSubSys2,
}

#[derive(Debug, Clone)]
struct DummySpawner;

struct DummyCtx;

fn main() {
	let overseer = Overseer::<_,_>::builder()
		.sub0(AwesomeSubSys::default())
		.spawner(DummySpawner)
		.build(|| -> DummyCtx { DummyCtx } );
}
