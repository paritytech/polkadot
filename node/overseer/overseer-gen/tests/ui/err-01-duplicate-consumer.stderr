error[E0119]: conflicting implementations of trait `std::convert::From<MsgStrukt>` for type `AllMessages`
  --> $DIR/err-01-duplicate-consumer.rs:19:1
   |
19 | #[overlord(signal=SigSigSig, event=Event, gen=AllMessages, error=OverseerError)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `AllMessages`
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `polkadot_overseer_gen::SubsystemSender<MsgStrukt>` for type `OverseerSubsystemSender`
  --> $DIR/err-01-duplicate-consumer.rs:19:1
   |
19 | #[overlord(signal=SigSigSig, event=Event, gen=AllMessages, error=OverseerError)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `OverseerSubsystemSender`
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
